tools/fip_create/fip_create.c

```diff
@@ -584,6 +586,24 @@ static int parse_cmdline(int argc, char **argv, struct option *options,
 			}
 			break;
 
+		case OPT_PLAT_TOC_FLAGS:
+		{
+			long plat_toc_flags;
+
+			errno = 0;
+			plat_toc_flags = strtol(optarg, NULL, 0);
+			if ((plat_toc_flags > UINT16_MAX) || (plat_toc_flags < 0)) {
+				printf("plat_toc_flags max is 16 bits: %s\n", optarg);
+				return -1;
+			}
+			if (errno) {
+				printf("plat_toc_flags parsing error: %s\n", optarg);
+				return -1;
+			}
+			*toc_flags |= plat_toc_flags << 32;
```

Comment:
32 is a magic number and you could do with a comment explaining that plat_toc_flags is a 16 bit field at bits 32-47. Also, a 16-bit mask of plat_toc_flags before the |= would be good.